internal/service/alarms/internal/dictionary/dictionary.go

```diff
@@ -0,0 +1,187 @@
+	wg := sync.WaitGroup{}
+	resultChannel := make(chan result)
+	for _, resource := range resourceTypeList {
+		wg.Add(1)
+		go func(resource resourceType) {
+			var err error
+			var rules []*monitoringv1.PrometheusRule
+
+			defer func() {
+				wg.Done()
+				resultChannel <- result{
+					resourceTypeID: resource.id,
+					rules:          rules,
+					err:            err,
+				}
+			}()
+
+			switch resource.model {
+			case resourceTypeCluster:
+				rules, err = r.getRulesForCluster(ctx, resource.version)
+			default:
+				err = fmt.Errorf("unsupported resource type: %s", resource.model)
+			}
+		}(resource)
+	}
+
+	go func() {
+		wg.Wait()
+		close(resultChannel)
+	}()
+
+	for res := range resultChannel {
+		if res.err != nil {
+			slog.Error("error fetching rules for resource type", "ResourceType ID", res.resourceTypeID, "error", res.err)
+			continue
+		}
+
+		r.RulesMap[res.resourceTypeID] = res.rules
+		slog.Info("loaded rules for resource type", "ResourceType ID", res.resourceTypeID, "rules count", len(res.rules))
+	}
+
+	// TODO: Load future alarm dictionary struct
+	// TODO: Load alarm dictionary into DB
+}
+
+// getRulesForCluster fetches Prometheus rules for a cluster resource type
+func (r *AlarmDictionary) getRulesForCluster(ctx context.Context, version string) ([]*monitoringv1.PrometheusRule, error) {
+	cluster, err := r.getManagedCluster(ctx, version)
+	if err != nil {
+		return nil, err
+	}
+
+	rules, err := r.getRules(ctx, cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	return rules.Items, nil
+}
+
+// getManagedCluster finds a single managed cluster with the given version
+func (r *AlarmDictionary) getManagedCluster(ctx context.Context, version string) (*clusterv1.ManagedCluster, error) {
+	// Match managed cluster with the given version and not local cluster
+	selector := labels.NewSelector()
+	versionSelector, _ := labels.NewRequirement(managedClusterVersionLabel, selection.Equals, []string{version})
+	localClusterRequirement, _ := labels.NewRequirement(localClusterLabel, selection.NotEquals, []string{"true"})
+
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, clients.ListRequestTimeout)
+	defer cancel()
+
+	var managedClusters clusterv1.ManagedClusterList
+	err := r.Client.List(ctxWithTimeout, &managedClusters, &client.ListOptions{
+		LabelSelector: selector.Add(*versionSelector).Add(*localClusterRequirement),
+		Limit:         1,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error listing managed clusters: %w", err)
+	}
+
+	if len(managedClusters.Items) == 0 {
+		return nil, fmt.Errorf("no managed cluster found with version %s", version)
+	}
+
+	return &managedClusters.Items[0], nil
+}
+
+var getClientForCluster = clients.NewClientForCluster
+
+// getRules gets Prometheus rules from a managed cluster
+func (r *AlarmDictionary) getRules(ctx context.Context, cluster *clusterv1.ManagedCluster) (*monitoringv1.PrometheusRuleList, error) {
```

Comment:
I would update the name to say "getPromRules". Cause the actual rules (a.k.a alarms definition) embedded in the spec under .spec.groups[].rules`

Reply:
Ack, thanks.

Reply:
Done